### PR TITLE
Rella 106 start turn method

### PR DESCRIPTION
**Pull request by rllzrc**
Start refactor base structure for StartTurn method, feedback greatly appreciated esp for setting round winner ~

### CODE REVIEW

File: WheelOfFortune/Game.cs
Comment (cychoi00) [inline WheelOfFortune/Game.cs:114]: So for this function, the expected input is of type `Action.ActionType` which is our enum in action. The different ways you can access this type is either by using `Action.ActionType.SolvePuzzleAction`/`Action.ActionType.GuessLetterAction` OR you can access them with numbers (think kind of like indexes in js) in this kind of format `(Action.ActionType)0`/`(Action.ActionType)1`. This may require us to either typecast the response or update the GetActionType to return an int value

File: WheelOfFortune/Game.cs
Comment (cychoi00) [inline WheelOfFortune/Game.cs:136]: Here you want to set your `isPuzzleSolved` variable to the return of `CurrentPlayer.CurrentAction.Execute` on line 121 (although it may be better to name it something like `isGuessCorrect` or maybe even `continuePlayerTurn`). If the value is true, then you want to check if the action that the player was taking was the GuessLetterAction or the SolvePuzzleAction, which you could do either by using the ActionTypeProperty that Safy recently added (`CurrentPlayer.CurrentAction.ActionTypeProperty == Action.ActionType.SolvePuzzleAction`) OR by checking the type of the action (`CurrentPlayer.CurrentAction.GetType() == typeof(SolvePuzzleAction` <- this may need to be tested, as I've never actually written this snippet of code before). I think I might have confused you when I mentioned `typeof` earlier, but you can just check that the `isPuzzleSolved` variable is true or false (or just use the variable itself since it's a boolean).

**Comment by cychoi00** [comment]
Left some comments - let me know if you want to discuss them! Also, it would be good practice to remove any unnecessary comments before you merge, but we can also do a final code cleanup later too 😸

File: WheelOfFortune/Game.cs
Comment (rllzrc) [reply WheelOfFortune/Game.cs:114]: Ah gotcha! I remember seeing this similarly in the Prompt class for the PromptAction method. I wasn't sure if it was this or if I should be using the return value from Prompt.GetActionType which is why I passed in the stored variable. I believe it is currently returning a string so I can cast it to return an int instead.

File: WheelOfFortune/Game.cs
Comment (rllzrc) [reply WheelOfFortune/Game.cs:136]: Oh good naming convention call out! I was totally confusing myself with the old implementation and forgot that we do not have the CurrentPlayer.Perform action method anymore. Hahaha yes, I was also confused by the typeof comment earlier since I was passively listening in on the teams convo as I was coding this out. I'll also clean up the comments prior to merging ~

**Review by cychoi00 — APPROVED** [review]
good to merge after you add updates from master!

**Comment by rllzrc** [comment]
😷 🐹 💇 GG !! ~

**Pull request merged**
